Code/Framework/AzFramework/AzFramework/Viewport/ViewportBus.h

```diff
@@ -77,4 +77,31 @@ namespace AzFramework
     };
 
     using ViewportBorderRequestBus = AZ::EBus<ViewportBorderRequests>;
+
+    //! Provides a bus to notify about different states of the ImGui.
+    class ViewportImGuiNotifications : public AZ::EBusTraits
+    {
+    public:
+        //! Notification that the ImGui menu bar has been activated.
+        virtual void OnImGuiActivated()
+        {
+        }
+
+        //! Notification that the ImGui menu bar has been deactivated.
+        virtual void OnImGuiDeactivated()
+        {
+        }
+
+        //! Notification that ImGui menu dropdown is activated.
+        virtual void OnImGuiDropDownDisplayed()
```

Comment:
nit: Shown is a good opposite to Hidden that's quite common, consider changing `Displayed` to this